src/components/views/rooms/RoomSettings.js

```diff
@@ -56,15 +56,15 @@ module.exports = React.createClass({
             tags_changed: false,
             tags: tags,
             areNotifsMuted: areNotifsMuted,
+            isRoomPublished: false, // updated in componentWillMount
```

Comment:
This will race with getting the room directory visibility. We probably want to guard against this (of which the simplest would probably be to set this to `undefined` and then not touch the visibility if it is still `undefined` on save).